### ISSUE TITLE
Doesn’t work with Variable Fonts?

### ISSUE THREAD

**Issue by paperboyo** [googlefonts/fontcrunch#15]
Hello,

Thank you for the great tool!

Using fonttools 3.19.0, when trying to crunch a Variable Font, maybe predictably as nodes per-master are not harmonised any more (?), I’m getting errors in the `gvar` table:

```
ERROR: An exception occurred during the decompilation of the 'gvar' table
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\fontTools\ttLib\__init__.py", line 426, in __getitem__
    table.decompile(data, self)
  File "C:\Python27\lib\site-packages\fontTools\ttLib\tables\_g_v_a_r.py", line 111, in decompile
    numPointsInGlyph, sharedCoords, axisTags, gvarData)
  File "C:\Python27\lib\site-packages\fontTools\ttLib\tables\_g_v_a_r.py", line 229, in decompileGlyph_
    sharedTuples, data, 4, offsetToData)
  File "C:\Python27\lib\site-packages\fontTools\ttLib\tables\TupleVariation.py", line 554, in decompileTupleVariationStore
    tableTag, axisTags, tupleData, pointDeltaData))
  File "C:\Python27\lib\site-packages\fontTools\ttLib\tables\TupleVariation.py", line 597, in decompileTupleVariation_
    len(points), tupleData, pos)
  File "C:\Python27\lib\site-packages\fontTools\ttLib\tables\TupleVariation.py", line 433, in decompileDeltas_
    assert len(result) == numDeltas
AssertionError
Dumping 'gvar' table...
```
Is there any hope here? I haven’t found any other tool that maintains contour integrity while offering so much smaller files. Or, maybe, I’m doing something wrong as my understanding of all this is very, very thin...

Regards
m.

**Comment by davelab6** [comment]
Look on my works ye mighty and despair :) 

VF requires points are interpolation compatible, and if you run a few fonts through fontcrunch, they are likely to diverge from compatibility. You'd need to use a tool like Prepolator, GlyphsApp (in a future version with quadratic spline support) or FontLab 6 to determine where they were incompatible, exactly, and fix them.

**Comment by paperboyo** [comment]
Much learning stretches far away in front of me :)

Thank you for the pointers!

**Comment by behdad** [comment]
Use cu2qu on the input instead.

**Comment by anthrotype** [comment]
https://github.com/googlei18n/cu2qu

**Comment by paperboyo** [comment]
Thank you both kindly. The fonts I have been playing with have static family members with cubic outlines. Because of earlier experiments with CFF-flavoured woff2 being ever-so-slightly larger than TTF-flavoured ones, because my expertise stretches about as far as (blindly) operating FLVI and because FLVI supports only TTF-flavoured VFs, I have been converting them to quadratic.

Testing FLVI conversion at different tolerance settings, I have found out that fontcrunch is much more accurate at the same (node no.) efficiency (maybe because it doesn’t observe compatibility). But needing point compatibility, I will have a look at cu2qu!

Just as soon as I, naively, thought I have tried [almost everything to get the fonts smaller](https://gist.github.com/paperboyo/f5809901b044cd49d7e7ada140fa1584), the whole new world of possibilities (and decisions) opens up with VFs... I blame [Erdoğan](https://github.com/guardian/frontend/pull/9411)!
